contracts/components/scanners/ScannerToNodeRunnerMigration.sol

```diff
@@ -0,0 +1,140 @@
+import "./IScannerMigration.sol";
+import "../node_runners/NodeRunnerRegistry.sol";
+
+/**
+ * Migration of ScannerRegistry to NodeRunnerRegistry
+ */
+contract ScannerToNodeRunnerMigration is BaseComponentUpgradeable, IScannerMigration {
+    /** Contract version */
+    string public constant version = "0.1.0";
+    uint256 public constant NODE_RUNNER_NOT_MIGRATED = 0;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ScannerRegistry public immutable scannerNodeRegistry;
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    NodeRunnerRegistry public immutable nodeRunnerRegistry;
+    uint256 public migrationEndTime;
+
+    event SetMigrationEndtime(uint256 migrationEndTime);
+    event MigrationExecuted(uint256 scannersMigrated, uint256 scannersIgnored, uint256 indexed nodeRunnerId, bool mintedNodeRunner);
+
+    error NotOwnerOfNodeRunner(address pretender, uint256 nodeRunnerId);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _forwarder,
+        address _scannerNodeRegistry,
+        address _nodeRunnerRegistry
+    ) initializer ForwardedContext(_forwarder) {
+        if (_scannerNodeRegistry == address(0)) revert ZeroAddress("_scannerNodeRegistry");
+        if (_nodeRunnerRegistry == address(0)) revert ZeroAddress("_nodeRunnerRegistry");
+        scannerNodeRegistry = ScannerRegistry(_scannerNodeRegistry);
+        nodeRunnerRegistry = NodeRunnerRegistry(_nodeRunnerRegistry);
+    }
+
+    /**
+     * @notice Initializer method, access point to initialize inheritance tree.
+     * @param __manager address of AccessManager.
+     * @param __migrationEndTime time when migration period ends
+     */
+    function initialize(address __manager, uint256 __migrationEndTime) public initializer {
+        __BaseComponentUpgradeable_init(__manager);
+        _setMigrationEndTime(__migrationEndTime);
+    }
+
+    /**
+     * @notice Method to self migrate from the old ScannerRegistry NFTs to a single NodeRunnerRegistry NFT.
+     * WARNING: ScannerNodeRegistry's manager addresses will not be migrated, please user NodeRunnerRegistry's methods to set them again.
+     * @param scanners array of scanner addresses to be migrated.
+     * All the enabled (disabled flags set to 0) ScannerRegistry ERC721 identified by the uint256(address) in the input array will be:
+     * - Registered in NodeRunnerRegistry to the nodeRunnerId either indicated or generated, with the same chainId and metadata.
+     * - Deleted in ScannerNodeRegistry. The ERC721 will be burned, disabled flags and managers deleted from storage.
+     * Scanners with disabled flags != 0 will be ignored (opted out), and will stay disabled in ScannerNodeRegistry.
+     * @param nodeRunnerId If the set as 0, the NodeRunnerRegistry ERC721 will be minted to the sender (address must not own any),
+     * set as a valid NodeRunnerRegistry ERC721 id owned by sender address otherwise.
```

Comment:
Same suggestion as the previous one, but it wasn't changed here.

```suggestion
     * @param nodeRunnerId If set as 0, a new NodeRunnerRegistry ERC721 will be minted to nodeRunner (but it must not own any prior),
     * otherwise must be set as a valid NodeRunnerRegistry ERC721 id owned by nodeRunner.
```